### PR TITLE
Clean up some arm64 prolog/epilog code

### LINKED ISSUE #22056: [arm64] Are unwind codes after frame pointer established unnecessary?
In some arm64 frame types, we establish the frame pointer and then subtract off the remainder of the frame, e.g.:
```
stp     fp, lr, [sp,#-16]!
mov     fp, sp
sub     sp, sp, #0x910
```
And this `sub sp` is reported in the unwind info. Does it need to be? Once the frame pointer is established, the unwinder should be able to unwind the callee-saved regs and the rest of the frame.

### PR DESCRIPTION
**Pull request by BruceForstall**
1. For frame types which establish a frame pointer before the final
SP adjustment, there is no need to report the SP adjustment in unwind
data, as it is unused.
2. Added some comments, esp. related to frameType 1, explaining the
totalFrameSize limit.
3. Fixed frameType 3 #outsz check to > 504 versus >= 504; 504 is a legal
offset for STP/LDP.
4. Fix frameType 3 epilogs to always restore SP from FP. Should give more
prolog/epilog unwind code matching, and simplifies the code.

Closes #22056, #22058, #22057, #22476, #22478.

### CODE REVIEW

**Comment by BruceForstall** [comment]
@dotnet-bot test Ubuntu16.04 arm64 Cross Checked no_tiered_compilation Build and Test
@dotnet-bot test Ubuntu16.04 arm64 Cross Checked jitstress1 Build and Test

**Comment by BruceForstall** [comment]
@dotnet-bot test Ubuntu16.04 arm64 Cross Checked gcstress0xc Build and Test

**Comment by BruceForstall** [comment]
@dotnet-bot test Ubuntu16.04 arm64 Cross Checked no_tiered_compilation Build and Test
@dotnet-bot test Ubuntu16.04 arm64 Cross Checked jitstress1 Build and Test
@dotnet-bot test Ubuntu16.04 arm64 Cross Checked gcstress0xc Build and Test

**Comment by BruceForstall** [comment]
Diffs:

Found 52 files with textual diffs.
Crossgen Diffs for System.Private.CoreLib.dll, framework assemblies, assemblies in F:\gh\coreclr10\bin\tests\Windows_NT.x64.Checked for x64 protononjit.dll
Summary:
(Lower is better)
Total bytes of diff: -8 (0.00% of base)
    diff is an improvement.
Top file improvements by size (bytes):
          -8 : JIT\Regression\CLR-x86-JIT\V1.2-Beta1\b191926\b191926\b191926.dasm (-0.01% of base)
1 total files with size differences (1 improved, 0 regressed), 8814 unchanged.
Top method improvements by size (bytes):
          -8 (-0.01% of base) : JIT\Regression\CLR-x86-JIT\V1.2-Beta1\b191926\b191926\b191926.dasm - f8192:Main():int (2 methods)
Top method improvements by size (percentage):
          -8 (-0.01% of base) : JIT\Regression\CLR-x86-JIT\V1.2-Beta1\b191926\b191926\b191926.dasm - f8192:Main():int (2 methods)
1 total methods with size differences (1 improved, 0 regressed), 388500 unchanged.
19 files had text diffs but not size diffs.
JIT\Methodical\fp\exgen\10w5d_cs_d\10w5d_cs_d.dasm had 448 diffs
JIT\Methodical\fp\exgen\10w5d_cs_r\10w5d_cs_r.dasm had 448 diffs
JIT\Directed\perffix\primitivevt\mixed1_cs_d\mixed1_cs_d.dasm had 96 diffs
JIT\Directed\perffix\primitivevt\mixed1_cs_r\mixed1_cs_r.dasm had 96 diffs
JIT\Regression\JitBlue\GitHub_17777\GitHub_17777\GitHub_17777.dasm had 82 diffs
Completed analysis in 204.59s

There are some unwind info improvements, where we now match prolog/epilog. The codegen diffs are because we restore SP from FP in the epilog instead of adjusting SP by a constant.

**Comment by BruceForstall** [comment]
@sandreenko PTAL
cc @dotnet/jit-contrib

**Comment by CarolEidt** [comment]
@BruceForstall - did you run the diffs natively on Arm64? I've had some difficulty with that in the past, but haven't tried lately.

File: src/jit/codegencommon.cpp [hunk 1]
```diff
@@ -4962,17 +4966,18 @@ void          CodeGen::genPushCalleeSavedRegisters()
...
         {
             // Case #1.
             //
             // Generate:
             //      stp fp,lr,[sp,#-framesz]!
             //
-            // The (totalFrameSize < 512) condition ensures that both the predecrement
-            // and the postincrement of SP can occur with STP.
+            // The (totalFrameSize <= 504) condition ensures that both the pre-index STP instruction
+            // used in the prolog, the post-index LDP instruction used in the epilog, can be generated.
```
Comment (CarolEidt) [inline src/jit/codegencommon.cpp]: i think this needs an "and" before "the post-index"

**Comment by BruceForstall** [comment]
> did you run the diffs natively on Arm64?

no -- I ran crossgen altjit diffs on Windows x64

**Comment by CarolEidt** [comment]
> no -- I ran crossgen altjit diffs on Windows x64

Thanks, I was just hoping :-)

**Comment by BruceForstall** [comment]
@CarolEidt note that there is a pmi_asm_diffs triggerable task on Linux arm64, although it doesn't quite work right now due to being configured to use `zip` which apparently no longer (?) exists on these machines. But it might work to use it manually -- the diffs themselves seemed to work.

**Pull request merged**